shootout/shootout.py

```diff
@@ -0,0 +1,173 @@
+            )
+        )
+
+        def check(m):
+            return m.content.lower() == listen_message.lower() and m.channel == ctx.channel and m.author in players
+
+        message = await self.bot.wait_for("message", check=check)
+        winner = message.author
+        await self.end_game(ctx, winner)
+
+    async def end_game(self, ctx, winner: discord.Member):
+        currency = await bank.get_currency_name(ctx.guild)
+        total = await self.config.guild(ctx.guild).Session.Pot()
+        try:
+            await bank.deposit_credits(winner, total)
+        except BalanceTooHigh as e:
+            await bank.set_balance(winner, e.max_balance)
+        await ctx.send(
+            "{} has won the shootout! {} {} has been deposited to their account!".format(
+                winner.mention, total, currency
+            )
+        )
+        await self.config.guild(ctx.guild).Session.clear()
+
+    @commands.command()
+    @commands.guild_only()
+    async def shootout(self, ctx):
+        """Start or join a shootout!
+
+        A shootout is a game where players can join by putting up money into the pot (if configured).
+
+        The bot will send a message to chat saying to begin and what message to type to win.
+
+        First person who types the message wins the entire pot! (Even if that pot is 0).
+
+        The game will not start if no other players have joined, and any cost will be refunded.
+
+        There is no maximum number of players.
+        """
+        settings = await self.config.guild(ctx.guild).all()
+        if await self.game_checks(ctx, settings):
+            await self.add_player(ctx, settings["cost"])
+
+    @commands.group()
+    @commands.guild_only()
+    @checks.mod_or_permissions(manage_guild=True)
+    async def soset(self, ctx):
+        """Change settings to Shootout"""
+        pass
+
+    @soset.command(name="bet")
+    async def soset_bet(self, ctx, amount: int):
+        """Sets the bet amount for a shootout"""
+        if amount < 0:
+            return await ctx.send("Get your negativity out of here!")
+        await self.config.guild(ctx.guild).cost.set(amount)
+        await ctx.tick()
+        return
+
+    @soset.command(name="message")
+    async def soset_message(self, ctx, message: str):
+        """Sets the message that the bot listens for during shootout games. Only affects new games!"""
+        await self.config.guild(ctx.guild).message.set(message)
+        await ctx.tick()
+        return
+
+    @soset.command(name="wait")
+    async def soset_wait(self, ctx, wait: int):
+        """Sets the wait time until the shootout starts after player 1 runs the command."""
+        if wait < 0:
+            return await ctx.send("Get your negativity out of here!")
+        await self.config.guild(ctx.guild).wait_time.set(wait)
+        await ctx.tick()
+
+    @commands.guild_only()
+    @checks.admin_or_permissions(administrator=True)
+    @commands.command(hidden=True)
```

Comment:
I wouldn't make this hidden, should be seen by admins who run help command.

Reply:
Debug config dump and reset commands are now visible.